### ISSUE TITLE
Incremental recap dump files are not uploaded to S3 

### ISSUE THREAD

**Issue by eliotjordan** [pulibrary/bibdata#1336]
Incremental recap dump files are not being uploaded to S3 correctly. In a republishing job with 26 dump files, only one file is uploaded to the S3 bucket.

**Comment by eliotjordan** [comment]
This looks to be a performance & memory issue. On the alma staging machine (5 sidekiq threads), this is the data from `top`:

25 minutes:   99.3% cpu  73.7% memory
40 minutes:   99.3% cpu  90.0% memory

Best guess, the sidekiq jobs saturate the available memory. It's strange and annoying that the jobs are considered "complete" when the threads terminate.

`perf` wasn't that helpful

```
   5.60%  util.rb:23     libruby-2.6.so.2.6.6  [.] 0x00000000000ae361
   5.11%  util.rb:23     libruby-2.6.so.2.6.6  [.] 0x00000000000ae358
   3.52%  util.rb:23     libruby-2.6.so.2.6.6  [.] 0x00000000000ae33f
   2.85%  util.rb:23     libc-2.27.so          [.] 0x00000000000bb4ea
   2.77%  util.rb:23     libc-2.27.so          [.] 0x00000000000bb4e6
   2.77%  util.rb:23     libc-2.27.so          [.] 0x00000000000bb4f2
   2.70%  util.rb:23     libruby-2.6.so.2.6.6  [.] 0x00000000000b124d
   2.62%  util.rb:23     libc-2.27.so          [.] 0x00000000000bb4ee
   2.27%  util.rb:23     libruby-2.6.so.2.6.6  [.] rb_wb_protected_newobj_of
   2.08%  util.rb:23     libc-2.27.so          [.] cfree
   1.55%  util.rb:23     libruby-2.6.so.2.6.6  [.] 0x00000000000b1154
   1.52%  util.rb:23     libruby-2.6.so.2.6.6  [.] 0x00000000000b2488
   1.47%  util.rb:23     libruby-2.6.so.2.6.6  [.] 0x00000000000b1e14
   1.45%  util.rb:23     libruby-2.6.so.2.6.6  [.] 0x00000000000af172
   1.25%  util.rb:23     libc-2.27.so          [.] malloc
```

**Comment by tpendragon** [comment]
It's not that they're considered complete, it's that the OS terminates the thread to free memory and the job doesn't make it back. Sidekiq-pro fixes that.

**Comment by eliotjordan** [comment]
"If a Sidekiq process crashes while processing a job, that job is lost." Yep. That's what's going on. Not sure if it's worth $950 / year.

**Comment by eliotjordan** [comment]
Closed by #1350
